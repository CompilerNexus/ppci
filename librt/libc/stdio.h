#ifndef STDIO_H
#define STDIO_H

void printf(char*);
void bsp_putc(char);

#endif

