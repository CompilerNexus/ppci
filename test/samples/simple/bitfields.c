
// Demonstrates bitfields
// See also: https://en.cppreference.com/w/c/language/bit_field
//

#include <stdio.h>

struct S {
 // three-bit unsigned field,
 // allowed values are 0...7
 unsigned int b : 3;
 unsigned int c : 12;
};

int main_main(void)
{
    struct S s = {7, 42};
    printf("s.b=%d\n", (int)s.b); // output: 7
    ++s.b; // unsigned overflow
    printf("s.b=%d\n", (int)s.b); // output: 0
    printf("s.c=%d\n", (int)s.c); // output: 42
    s.c++;
    printf("s.c=%d\n", (int)s.c); // output: 43
}

