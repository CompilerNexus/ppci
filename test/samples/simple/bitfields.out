u.b=7
u.b=0
u.c=42
u.c=43
s.b=3
s.b=-4
s.c=7
s.c=-8
s.d=1
s.d=-2
